### PR TITLE
62 create parent table

### PR DESCRIPTION
**Pull request by AlaaTaima**
Relates #62

### CODE REVIEW

File: client/src/Components/CommonComponent/TableParent/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { BrowserRouter as Router, Link } from 'react-router-dom';
+// import data from './data';
+import './index.css';
+
+const Table = data => {
```
Comment (11janci) [inline client/src/Components/CommonComponent/TableParent/index.js]: same as the other one, the name is misleading, suggesting a generic table component (the 'parent' relationship is also not clear to me). It should be sth like 'StudentTable' or 'StudentList'.

File: client/src/Components/CommonComponent/TableParent/index.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                <td className="parentTable__content1">{e.name}</td>
+                <td className="parentTable__content2">{e.grade}</td>
+                <td className="parentTable__content2">
+                  {e.class}
+                  <p>
+                    <Link
+                      to={`/student/:${e.id}`}
+                      className="parentTable__link"
+                    >
+                      view subject
```
Comment (11janci) [inline client/src/Components/CommonComponent/TableParent/index.js]: shouldn't this be 'view student' / 'view profile'?

File: client/src/Components/CommonComponent/Table/index.js [hunk 3]
```diff
@@ -13,36 +13,34 @@ const Table = (data, id) => {
...
-          );
-        })}
+        {data.map(e => (
+          <tr>
+            <Router>
+              <td className="studentTable__content1">
+                {e.name}
+                <p>
+                  <Link
+                    to={`/profile/teacher/:${e.id}`}
```
Comment (11janci) [inline client/src/Components/CommonComponent/Table/index.js]: btw, I know this is not your changes, but it is good to have these URLs defined in a separate file as constants/functions instead of hardcoded strings like the one here, so that in case the URL changes in the future, you have just one place to update and don't have to search through the whole codebase...

File: client/src/Components/CommonComponent/Table/index.js [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { BrowserRouter as Router, Link } from 'react-router-dom';
 // import data from './data';
 import './index.css';
 
-const Table = (data, id) => {
+const Table = data => {
```
[the remark this answers is not in the document]
Comment (AlaaTaima) [reply client/src/Components/CommonComponent/Table/index.js]: thanks, I will edit it.

File: client/src/Components/CommonComponent/Table/index.js (hunk 3 above)
Comment (AlaaTaima) [reply client/src/Components/CommonComponent/Table/index.js]: OK, I will try your suggestion.
thanks

File: client/src/Components/CommonComponent/TableParent/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlaaTaima) [reply client/src/Components/CommonComponent/TableParent/index.js]: OK, thanks for your comment.

File: client/src/Components/CommonComponent/TableParent/index.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+          <th className="parentTable__title2">Class</th>
+        </tr>
+      </thead>
+      <tbody className="parentTable__content">
+        {data.map(e => {
+          return (
+            <tr>
+              <Router>
+                <td className="parentTable__content1">{e.name}</td>
+                <td className="parentTable__content2">{e.grade}</td>
```
[the remark this answers is not in the document]
Comment (AlaaTaima) [reply client/src/Components/CommonComponent/TableParent/index.js]: ok

File: client/src/Components/CommonComponent/TeacherTable/index.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+                  >
+                    view profile
+                  </Link>
+                </p>
+              </td>
+              <td className="studentTable__content2">
+                {subject}
+                <p>
+                  <Link
+                    to={`/student/:subject/:${id}	`}
```
Comment (aminalakhsham) [inline client/src/Components/CommonComponent/TeacherTable/index.js]:
```suggestion
                    to={`/student/:subject/:${id}`}
```

File: client/src/Components/CommonComponent/StudentTable/data.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
+// eslint-disable-next-line import/no-mutable-exports
```
Comment (aminalakhsham) [inline client/src/Components/CommonComponent/StudentTable/data.js]: remove this and fix eslint error

File: client/src/Components/CommonComponent/StudentTable/index.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { BrowserRouter as Router, Link } from 'react-router-dom';
```
Comment (aminalakhsham) [inline client/src/Components/CommonComponent/StudentTable/index.js]:
```suggestion
import { Link } from 'react-router-dom';
```

File: client/src/Components/CommonComponent/StudentTable/index.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+          <th className="parentTable__title1">Name</th>
+          <th>Grade</th>
+          <th className="parentTable__title2">Class</th>
+        </tr>
+      </thead>
+      <tbody className="parentTable__content">
+        {data.map(({ name, grade, StudentClass, id }) => {
+          return (
+            <tr>
+              <Router>
```
Comment (aminalakhsham) [inline client/src/Components/CommonComponent/StudentTable/index.js]:
```suggestion

```

File: client/src/Components/CommonComponent/TeacherTable/index.js [hunk 10]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { BrowserRouter as Router, Link } from 'react-router-dom';
```
Comment (aminalakhsham) [inline client/src/Components/CommonComponent/TeacherTable/index.js]:
```suggestion
import { Link } from 'react-router-dom';
```

File: client/src/Components/CommonComponent/TeacherTable/index.js [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+      <thead className="studentTable__title">
+        <tr>
+          <th className="studentTable__title1">Teacher Name</th>
+          <th className="studentTable__title2">Subject</th>
+        </tr>
+      </thead>
+      <tbody className="studentTable__content">
+        {data.map(({ teacherName, subject, id }) => (
+          <tr>
+            <Router>
```
Comment (aminalakhsham) [inline client/src/Components/CommonComponent/TeacherTable/index.js]:
```suggestion

```

File: client/src/Components/CommonComponent/TeacherTable/index.js [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+                >
+                  view profile
+                </Link>
+              </p>
+            </td>
+            <td className="studentTable__content2">
+              {subject}
+              <p>
+                <Link
+                  to={`/student/:subject/:${id}	`}
```
Comment (aminalakhsham) [inline client/src/Components/CommonComponent/TeacherTable/index.js]:
```suggestion
                  to={`/student/:subject/:${id}`}
```

File: client/src/Components/CommonComponent/StudentTable/index.js [hunk 13]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import './index.css';
+
+const StudentTable = data => {
+  return (
+    <table className="parentTable">
```
Comment (11janci) [inline client/src/Components/CommonComponent/StudentTable/index.js]: thanks for the changes! 👍 could you pls make another PR and update all of the CSS classnames to `studentTable`?

File: client/src/Components/CommonComponent/TeacherTable/index.js [hunk 14]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import './index.css';
+
+const TeacherTable = data => {
+  return (
+    <table className="studentTable">
```
Comment (11janci) [inline client/src/Components/CommonComponent/TeacherTable/index.js]: similarly, pls update the class names to `teacherTable`
